src/main/kotlin/no/nav/syfo/person/skjermingskode/SkjermingskodeService.kt

```diff
@@ -1,32 +1,56 @@
 package no.nav.syfo.person.skjermingskode
 
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
 import no.nav.syfo.client.pdl.PdlClient
 import no.nav.syfo.client.skjermedepersonerpip.SkjermedePersonerPipClient
 import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.person.api.domain.PersonInfo
 import no.nav.syfo.person.api.domain.Skjermingskode
 
 class SkjermingskodeService(
     private val skjermedePersonerPipClient: SkjermedePersonerPipClient,
     private val pdlClient: PdlClient,
 ) {
-    suspend fun hentBrukersSkjermingskode(
+    suspend fun hentSkjermingskodeForPersonidenter(
         callId: String,
-        personIdent: PersonIdentNumber,
+        personidenter: List<PersonIdentNumber>,
         token: String,
-    ): Skjermingskode? {
-        val hasAdressebeskyttelse = pdlClient.hasAdressebeskyttelse(callId = callId, personIdent = personIdent)
+    ) =
+        personidenter.map { personident ->
+            val skjermingskode = hentBrukersSkjermingskode(
+                callId = callId,
+                personident = personident,
+                token = token,
+            )
+            Pair(personident, skjermingskode)
+        }.map { (personident, skjermingskode) ->
+            PersonInfo(
+                fnr = personident.value,
+                skjermingskode = skjermingskode.await(),
+            )
+        }
 
-        return hasAdressebeskyttelse?.let {
-            when {
-                hasAdressebeskyttelse -> Skjermingskode.DISKRESJONSMERKET
-                else -> if (
-                    skjermedePersonerPipClient.isSkjermet(
+    private suspend fun hentBrukersSkjermingskode(
+        callId: String,
+        personident: PersonIdentNumber,
+        token: String,
+    ): Deferred<Skjermingskode> =
+        CoroutineScope(Dispatchers.IO).async {
```

Comment:
Kan være vi støter på de samme problemene vi har hatt med norg2 i en annen sammenhent. Kanskje vi kan bruke en annen dispatcher som begrenser antall samtidige kall til feks 10 eller 20?

Reply:
Vet vi egentlig at det er samtidige kall som gjør at NORG2 feiler? 🤔 Fra [eksempelet i morres](https://logs.adeo.no/app/discover#/view/a90cb590-5515-11ed-b3e8-d969437dd878?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2024-10-16T05:40:00.000Z',to:'2024-10-16T05:50:00.000Z'))&_a=(columns:!(message,envclass,level,application,host),dataSource:(dataViewId:'96e648c0-980a-11e9-830a-e17bbd64b4db',type:dataView),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'96e648c0-980a-11e9-830a-e17bbd64b4db',key:envclass,negate:!f,params:(query:p),type:phrase),query:(match_phrase:(envclass:p))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'96e648c0-980a-11e9-830a-e17bbd64b4db',key:level,negate:!f,params:(query:Error),type:phrase),query:(match_phrase:(level:Error))),('$state':(store:appState),meta:(alias:!n,disabled:!f,field:team,index:'96e648c0-980a-11e9-830a-e17bbd64b4db',key:team,negate:!f,params:(query:teamsykefravr),type:phrase),query:(match_phrase:(team:teamsykefravr)))),hideChart:!f,interval:auto,query:(language:lucene,query:''),sort:!(!('@timestamp',desc)))) ser det ut som det egentlig bare er én henting som feiler.
Hvis tilfellet er at det er for mange samtidige kall så ville vi vel heller fått timeout? Om vi faktisk tar ned API'et til NORG2 så ville vi vel fått beskjed? Det kan jo selvfølgelig være at de har andre mekanismer her som gjør at vi får 503.

Reply:
I caset mot norg2 er det jo caching'en som gjør at det bare kommer sporadisk og om morgenen. Tror egentlig det er noe "på veien" som kaster 503 i det tilfellet (dvs tror ikke kallene kommer til norg2 i det hele tatt).

Synes det med `limitedParallelism` ser lurt ut, og et bra pragmatisk kompromiss.

Reply:
norg2-feilen har vel ikke de store konsekvensene så vi kan jo se om det blir et problem her og i så fall legge på `limitedParallelism` her hvis det har en effekt på norg2-feilen ref https://github.com/navikt/istilgangskontroll/pull/108